src/libpriv/rpmostree-scripts.c

```diff
@@ -207,6 +207,54 @@ script_child_setup (gpointer opaque)
     err (1, "dup2(stderr)");
 }
 
+/* Print the output of a script, with each line prefixed with
+ * the script identifier (e.g. foo.post: bla bla bla).
+ */
+static gboolean
+dump_buffered_output (const char *prefix,
+                      GLnxTmpfile *tmpf,
+                      GError     **error)
+{
+  if (lseek (tmpf->fd, 0, SEEK_SET) < 0)
+    return glnx_throw_errno_prefix (error, "lseek");
+  g_autoptr(FILE) buf = fdopen (tmpf->fd, "r");
+  if (!buf)
+    return glnx_throw_errno_prefix (error, "fdopen");
+  tmpf->fd = -1;  /* Ownership of fd was transferred */
+
+  while (TRUE)
+    {
+      size_t len = 0;
+      ssize_t bytes_read = 0;
+      g_autofree char *line = NULL;
+      errno = 0;
+      if ((bytes_read = getline (&line, &len, buf)) == -1)
+        {
+          if (errno != 0)
+            return glnx_throw_errno_prefix (error, "getline");
+          else
+            break;
+        }
+      printf ("%s: %s", prefix, line);
+      if (bytes_read > 0 && line[bytes_read-1] != '\n')
+        fputc ('\n', stdout);
+    }
+
+  return TRUE;
+}
+
+/* Since it doesn't make sense to fatally error if printing output fails, catch
+ * any errors there and print.
+ */
+static void
+dump_buffered_output_noerr (const char *prefix,
+                            GLnxTmpfile *tmpf)
+{
```

Comment:
We should just return if `tmpf` is not even initialized here, right? (The pkglayering `stdout_is_journal()` case).

Reply:
Yep, fixup :arrow_down: